backend/src/services/permission/permission.service.ts

```diff
@@ -0,0 +1,28 @@
+import { permission as permissionRep } from '~/data/repositories/repositories';
+import { PermissionResponseDto } from '~/common/types/types';
+
+type Constructor = {
+  permissionRepository: typeof permissionRep;
+};
+
+class Permission {
+  #permissionRepository: typeof permissionRep;
+
+  constructor({ permissionRepository }: Constructor) {
+    this.#permissionRepository = permissionRepository;
+  }
+
+  async getPermissions(): Promise<PermissionResponseDto[]> {
```

Comment:
```suggestion
  async getAll(): Promise<PermissionsGetAllResponseDto[]> {
```

`PermissionsGetAllResponseDto`

should be like

```ts
type PermissionsGetAllItemResponseDto = {
          id: number,
        key: string,
        name: string,
}

type PermissionsGetAllResponseDto = {
  items: PermissionsGetAllItemResponseDto[]
}
```